test/jellyfish_web/socket_test.exs

```diff
@@ -0,0 +1,135 @@
+defmodule JellyfishWeb.SocketTest do
+  use ExUnit.Case
+
+  import ExUnit.CaptureLog
+
+  alias Jellyfish.Peer.WebRTC
+  alias Jellyfish.{Room, RoomService}
+
+  alias JellyfishWeb.{Endpoint, Socket}
+
+  @data "mediaEventData"
+
+  setup do
+    {:ok, room} = RoomService.create_room(10)
+    {:ok, room_pid} = RoomService.find_room(room.id)
+    {:ok, peer} = Room.add_peer(room_pid, WebRTC)
+
+    on_exit(fn ->
+      RoomService.delete_room(room.id)
+    end)
+
+    %{room_id: room.id, peer_id: peer.id, room_pid: room_pid}
+  end
+
+  describe "connecting" do
+    test "when credentials are valid", %{room_id: room_id, peer_id: peer_id, room_pid: room_pid} do
+      assert {:ok, state} = connect(%{"room_id" => room_id, "peer_id" => peer_id})
+      assert state.peer_id == peer_id
+      assert state.room_id == room_id
+
+      status =
+        room_pid
+        |> Room.get_state()
+        |> Map.get(:peers)
+        |> Map.get(peer_id)
+        |> Map.get(:status)
```

Comment:
You might consider using `get_in`  but this depends on what we return from `Room.get_state`.